File: src/main/java/mods/cybercat/gigeresque/client/entity/render/ChestbursterEntityRenderer.java
package mods.cybercat.gigeresque.client.entity.render;

import com.mojang.blaze3d.vertex.PoseStack;
import mod.azure.azurelib.renderer.GeoEntityRenderer;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import org.jetbrains.annotations.NotNull;

import mods.cybercat.gigeresque.client.entity.model.ChestbursterEntityModel;
import mods.cybercat.gigeresque.client.entity.render.feature.BusterBloodFeatureRenderer;
import mods.cybercat.gigeresque.common.entity.impl.classic.ChestbursterEntity;

public class ChestbursterEntityRenderer extends GeoEntityRenderer<ChestbursterEntity> {

    public ChestbursterEntityRenderer(EntityRendererProvider.Context ctx) {
        super(ctx, new ChestbursterEntityModel());
        this.shadowRadius = 0.1f;
        this.addRenderLayer(new BusterBloodFeatureRenderer(this));
    }

    @Override
    public void render(
        ChestbursterEntity entity,
        float entityYaw,
        float partialTicks,
        PoseStack stack,
        @NotNull MultiBufferSource bufferIn,
        int packedLightIn
    ) {
        float scaleFactor = 1.0f + ((entity.getGrowth() / entity.getMaxGrowth()) / 4.0f);
        stack.scale(scaleFactor, scaleFactor, scaleFactor);
        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn);
    }

    @Override
    protected float getDeathMaxRotation(ChestbursterEntity entityLivingBaseIn) {
        return 0;
    }
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/render/ChestbursterEntityRenderer.kt
package com.bvanseg.gigeresque.client.entity.render

import com.bvanseg.gigeresque.client.entity.model.ChestbursterEntityModel
import com.bvanseg.gigeresque.client.extensions.scale
import com.bvanseg.gigeresque.common.entity.impl.ChestbursterEntity
import net.minecraft.client.render.VertexConsumerProvider
import net.minecraft.client.render.entity.EntityRendererFactory
import net.minecraft.client.util.math.MatrixStack
import software.bernie.geckolib3.renderers.geo.GeoEntityRenderer

/**
 * @author Boston Vanseghi
 */
class ChestbursterEntityRenderer(context: EntityRendererFactory.Context) : GeoEntityRenderer<ChestbursterEntity>(
    context,
    ChestbursterEntityModel()
) {
    init {
        this.shadowRadius = 0.1f
    }

    override fun render(
        entity: ChestbursterEntity,
        entityYaw: Float,
        partialTicks: Float,
        stack: MatrixStack,
        bufferIn: VertexConsumerProvider,
        packedLightIn: Int
    ) {
        stack.scale(1.0f + ((entity.growth / entity.maxGrowth.toFloat()) / 4.0f))
        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn)
    }
}